Tests/JSONFactoryTests.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  JSONFactoryTests.swift
+//  Matrioska
+//
+//  Created by Joan Romano on 12/01/17.
+//  Copyright © 2017 runtastic. All rights reserved.
+//
+
+import XCTest
+import Quick
+import Nimble
+@testable import Matrioska
+
+class JSONFactoryTests: QuickSpec {
+    
+    let bundle = Bundle(for: JSONFactoryTests.self)
+    let jsonFileName = "app_structure"
+    
+    override func spec() {
+        
+        let jsonObject = try! JSONReader.jsonObject(from: jsonFileName, bundle: bundle)!
+        let json = structure(from: jsonObject)
+        
+        let tabBarFactory = ClusterFactory.TabBarClusterFactory()
+        let stackFactory = ClusterFactory.StackClusterFactory()
+        let buttonFactory = ButtonFactory()
+        let labelFactory = LabelFactory()
+        let tableViewFactory = TableViewFactory()
+        
+        let factories: [ComponentFactory] = [tabBarFactory, stackFactory, buttonFactory, labelFactory, tableViewFactory]
+        
+        describe("Register factories") {
+            
+            context("when no factories added") {
+                let jsonFactory = JSONFactory()
+                
+                it("has no factories") {
+                    expect(jsonFactory.factories.count).to(equal(0))
+                }
+            }
+            
+            context("when registering factories") {
+                let jsonFactory = JSONFactory()
+                let tabBarFactory = ClusterFactory.TabBarClusterFactory()
+                let stackFactory = ClusterFactory.StackClusterFactory()
+                jsonFactory.register(tabBarFactory)
+                jsonFactory.register(stackFactory)
+                
+                it("has the registered factories") {
+                    expect(jsonFactory.factories.count).to(equal(2))
+                    expect(jsonFactory.factories[tabBarFactory.typeName()]).toNot(beNil())
+                    expect(jsonFactory.factories[stackFactory.typeName()]).toNot(beNil())
+                }
+                
+                it("does not have non registered factories") {
+                    expect(jsonFactory.factories["foo"]).to(beNil())
+                    expect(jsonFactory.factories["bar"]).to(beNil())
+                }
+            }
+            
+        }
+        
+        describe("Produce") {
+            
+            it("throws an assertion when the JSON object does not have mandatory keys") {
+                let jsonFactory = JSONFactory()
+                let faultyProduceNoKeys = { _ = try! jsonFactory.produce(from: ["foo": "bar"]) }
+                let faultyProduceNoType = { _ = try! jsonFactory.produce(from: ["id": "bar"]) }
+                
+                expect(faultyProduceNoKeys()).to(throwAssertion())
```

Comment:
should not throw an assertion. throws an error. However it's now asserting because you are `try!`ing